src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -3940,15 +3962,16 @@ namespace FanCoilUnits {
         }
 
         if (UnitOn && QZnReq < (-1.0 * Small5WLoad) && CoolingLoad) {
-
-            mdot = 0.0;
-            SetComponentFlowRate(mdot,
-                                 FanCoil(FanCoilNum).HeatCoilFluidInletNode,
-                                 FanCoil(FanCoilNum).HeatCoilFluidOutletNodeNum,
-                                 FanCoil(FanCoilNum).HeatCoilLoopNum,
-                                 FanCoil(FanCoilNum).HeatCoilLoopSide,
-                                 FanCoil(FanCoilNum).HeatCoilBranchNum,
-                                 FanCoil(FanCoilNum).HeatCoilCompNum);
+            if (FanCoil(FanCoilNum).HCoilType_Num == HCoil_Water) {
+                mdot = 0.0;
+                SetComponentFlowRate(mdot,
+                                     FanCoil(FanCoilNum).HeatCoilFluidInletNode,
+                                     FanCoil(FanCoilNum).HeatCoilFluidOutletNodeNum,
+                                     FanCoil(FanCoilNum).HeatCoilLoopNum,
+                                     FanCoil(FanCoilNum).HeatCoilLoopSide,
+                                     FanCoil(FanCoilNum).HeatCoilBranchNum,
+                                     FanCoil(FanCoilNum).HeatCoilCompNum);
+            }
```

Comment:
Previously the *Fancoilunit* with for MultiSpeedFan capacity control method was designed for hot-water heating coil only. This caused fatal out when electric heating coil is used with this capacity control. This protection is needed to avoid fatal out. There are a handful places where this protect is needed. The defect file fatal out because of this problem.